pkg/rpc/rpc_handler.go

```diff
@@ -98,7 +146,29 @@ func RunRPCHandler(policyReconciler *controllers.PolicyEndpointsReconciler) erro
 		return errors.Wrap(err, "network policy agent: failed to listen to gRPC port")
 	}
 	grpcServer := grpc.NewServer()
-	rpc.RegisterNPBackendServer(grpcServer, &server{policyReconciler: policyReconciler, log: rpcLog})
+
+	// Connect to metadata cache service
+	serviceIP, err := utils.GetServiceIP(clientset, "metadata-cache-service", "default", rpcLog)
```

Comment:
We're making an assumption that aws-k8s-metadata service is available and we exit if we fail to find such a service on the cluster. We can't exit out of the process. We should make this optional (i.e.,) we print an enhanced log message if we find the service and if not we print the 5 tuple info of the flow (SrcIP/SrcPort/DstIP/DstPort/Protocol) like we do today.

Reply:
`metadata-cache-service` --> let's make this a const and maybe add an `aws` prefix to it.

Reply:
I see that we're creating a K8S client obj just to find the service IP of metadata cache service and there is no need to do it. We can just resolve the service name (DNS resolution) and that should provide us with service IP.